### PR TITLE
Option to skip domain name verification of servers

### PR DESCRIPTION
**Pull request by Fritiofhedstrom**
https://github.com/eclipse-zenoh/zenoh/issues/513

Add option "server_name_verification": Option<bool>. If set to true (default) servers' certificates must use their IP address or domain name as a common name in their ca-signed certificate in order to connect. If set to false then the servers will be trusted regardless of their common name, this is useful if the IP address of a Zenoh server is difficult to know in advance

Some small fixes on docs for zenoh-cfg-properties that were wrong.

### CODE REVIEW

**Comment by Mallets** [comment]
Thanks @Fritiofhedstrom for your PR! We need you to sign the [Eclipse Contribution Agreement](https://accounts.eclipse.org/user/login?destination=legal/eca/validation/167096) to be able to merge it.

In the meanwhile, I've requested @milyin to review the PR.

File: io/zenoh-links/zenoh-link-tls/src/lib.rs [hunk 1]
```diff
@@ -99,6 +101,12 @@ impl ConfigurationInspector<Config> for TlsConfigurator {
                 tls_client_certificate.into(),
             );
         }
+        if let Some(server_name_verification) = c.server_name_verification() {
```
Comment (p-avital) [inline io/zenoh-links/zenoh-link-tls/src/lib.rs:104]: you could shorten this to 
```rust
match c.server_name_verification() {
    Some(true) | None => todo!("insert true"),
    Some(false) => todo!("insert false"),
}
```

**Review by p-avital — CHANGES_REQUESTED** [review]
You have some clippy warnings to fix (seems to be your update of rustls surfaced some deprecations) :)

File: commons/zenoh-config/src/lib.rs [hunk 2]
```diff
@@ -282,6 +282,7 @@ validated_struct::validator! {
                     client_auth: Option<bool>,
                     client_private_key: Option<String>,
                     client_certificate: Option<String>,
+                    server_name_verification: Option<bool>
```
Comment (p-avital) [inline commons/zenoh-config/src/lib.rs:285]: Could you add some documentation on that one in `DEFAULT_CONFIG.json5`, highlighting the dangers associated by not checking that the certificate came from the expected IP/name?

**Review by p-avital — CHANGES_REQUESTED** [review]

File: io/zenoh-links/zenoh-link-tls/src/unicast.rs [hunk 3]
```diff
@@ -603,6 +604,18 @@ impl TlsClientConfig {
             client_auth = value.parse()?
         }
 
+        let server_name_verification: bool =
+            if let Some(value) = config.get(TLS_SERVER_NAME_VERIFICATION) {
+                value
+            } else {
+                TLS_SERVER_NAME_VERIFICATION_DEFAULT
+            }
+            .parse()?;
```
Comment (milyin) [inline io/zenoh-links/zenoh-link-tls/src/unicast.rs:613]: This could be written more shortly:
```Rust
let server_name_verification: bool = config.get(TLS_SERVER_NAME_VERIFICATION)
  .unwrap_or(TLS_SERVER_NAME_VERIFICATION_DEFAULT).parse()?;
```

**Comment by Mallets** [comment]
Please run `cargo fmt --all` to make the CI happy about the code format.

**Review by Mallets — APPROVED** [review]

**Review by p-avital — APPROVED** [review]
